packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -85,18 +80,16 @@ type SchemaDerivedData = {
 };
 
 type ServerState =
-  | { phase: 'initialized with schema'; schemaDerivedData: SchemaDerivedData }
-  | { phase: 'initialized with gateway'; gateway: GatewayInterface }
-  | { phase: 'starting'; barrier: Resolvable<void> }
+  | { phase: 'initialized'; schemaManager: SchemaManager }
   | {
-      phase: 'invoking serverWillStart';
+      phase: 'starting';
       barrier: Resolvable<void>;
-      schemaDerivedData: SchemaDerivedData;
+      schemaManager: SchemaManager;
     }
   | { phase: 'failed to start'; error: Error; loadedSchema: boolean }
```

Comment:
While you're here you can remove `loadedSchema` which hasn't been read since https://github.com/apollographql/apollo-server/commit/49a71e4e01a8dbdb5c7151d9467ad0c164470cb6

Reply:
Sounds good, I've removed `loadedSchema` from the server state.